### PR TITLE
Make Bridge method ReceiveHeaders public as part of network upgrade R…

### PR DESCRIPTION
**Pull request by josedahlquist**
…SKIP124

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/Bridge.java [hunk 1]
```diff
@@ -373,6 +373,21 @@ public void updateCollections(Object[] args)
...
+    }
+
+    public long receiveHeadersGetCost(Object[] args) {
+        // Old, private method fixed cost. Only applies before the corresponding RSKIP
+        if (!activations.isActive(ConsensusRule.RSKIP124)) {
+            return 22000L;
+        }
+
+        // Dynamic cost based on the number of headers
+        final int numberOfHeaders = ((Object[]) args[0]).length;
```
Comment (pmprete) [inline rskj-core/src/main/java/co/rsk/peg/Bridge.java]: Please check that args is not null, if it is return basic cost

File: rskj-core/src/test/java/co/rsk/peg/performance/ReceiveHeadersTest.java [hunk 2]
```diff
@@ -38,43 +41,139 @@
...
+        doReceiveHeaders("warmup", 100, 1, 0);
+        setQuietMode(false);
+        System.out.print("Done!\n");
+    }
 
     @Test
-    public void receiveHeaders() throws IOException {
-        final int minBtcBlocks = 1000;
-        final int maxBtcBlocks = 2000;
+    public void receiveHeadersSingleBlock() throws IOException {
```
Comment (lsebrie) [inline rskj-core/src/test/java/co/rsk/peg/performance/ReceiveHeadersTest.java]: Please remove exceptions when are not possible

File: rskj-core/src/test/java/co/rsk/peg/BridgeTest.java [hunk 3]
```diff
@@ -2298,6 +2301,136 @@ public void mineableMethodsDefinition() {
...
+        when(txMock.getReceiveAddress()).thenReturn(PrecompiledContracts.BRIDGE_ADDR);
+        when(txMock.getSender()).thenReturn(sender);
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        when(bridgeSupportMock.getRetiringFederation()).thenReturn(null);
+        when(bridgeSupportMock.getActiveFederation()).thenReturn(BridgeRegTestConstants.getInstance().getGenesisFederation());
+
+        Bridge bridge = PowerMockito.spy(new Bridge(PrecompiledContracts.BRIDGE_ADDR, constants, activationConfig));
+        PowerMockito.doReturn(bridgeSupportMock).when(bridge, "setup");
+        bridge.init(txMock, getGenesisBlock(), null, null, null, null);
```
Comment (lsebrie) [inline rskj-core/src/test/java/co/rsk/peg/BridgeTest.java]: Maybe in another PR would be nice to have a function or class to abstract Bridge and BridgeSupport creation for tests

Comment (josedahlquist) [reply rskj-core/src/test/java/co/rsk/peg/BridgeTest.java]: agreed, we will be doing that after we wrap up this release

**Pull request merged**